src/module/item/class/data.ts

```diff
@@ -1,47 +1,201 @@
-import { AttributeString, SaveType, SkillSlug } from "@actor/types.ts";
-import { ABCSystemData, ABCSystemSource } from "@item/abc/data.ts";
-import { BaseItemSourcePF2e, RarityTraitAndOtherTags } from "@item/base/data/system.ts";
-import { ZeroToFour } from "@module/data.ts";
+import type { AttributeString, SkillSlug } from "@actor/types.ts";
+import { ATTRIBUTE_ABBREVIATIONS } from "@actor/values.ts";
+import type { ModelPropsFromSchema, SourceFromSchema } from "@common/data/fields.mjs";
+import { ABCFeatureEntryField } from "@item/abc/data.ts";
+import { ItemSystemModel, ItemSystemSchema } from "@item/base/data/model.ts";
+import type { BaseItemSourcePF2e, ItemSystemSource, RarityTraitAndOtherTags } from "@item/base/data/system.ts";
+import { ProficiencyRankField, RarityField } from "@module/model.ts";
+import { LaxArrayField, RecordField, SlugField } from "@system/schema-data-fields.ts";
+import type { ClassPF2e } from "./document.ts";
+import fields = foundry.data.fields;
 
 type ClassSource = BaseItemSourcePF2e<"class", ClassSystemSource>;
 
-interface ClassSystemSource extends ABCSystemSource {
-    traits: RarityTraitAndOtherTags;
-    keyAbility: { value: AttributeString[]; selected: AttributeString | null };
-    hp: number;
-    perception: ZeroToFour;
-    savingThrows: Record<SaveType, ZeroToFour>;
-    attacks: ClassAttackProficiencies;
-    defenses: ClassDefenseProficiencies;
-    /** Starting proficiency in "spell attack rolls and DCs" */
-    spellcasting: ZeroToFour;
-    trainedSkills: {
-        value: SkillSlug[];
-        additional: number;
-    };
-    ancestryFeatLevels: { value: number[] };
-    classFeatLevels: { value: number[] };
-    generalFeatLevels: { value: number[] };
-    skillFeatLevels: { value: number[] };
-    skillIncreaseLevels: { value: number[] };
-    level?: never;
+/** A list of old skill abbreviations meant to allow the data models to be migrateable in the meantime. */
+function getCompatSkills(): SkillSlug[] {
+    return [
+        "acr",
+        "arc",
+        "ath",
+        "cra",
+        "dec",
+        "dip",
+        "itm",
+        "med",
+        "nat",
+        "occ",
+        "prf",
+        "rel",
+        "soc",
+        "ste",
+        "sur",
+        "thi",
+        ...Object.keys(CONFIG.PF2E.skills),
+    ] as unknown as SkillSlug[];
 }
```

Comment:
maybe we can save some lines here by defining the short names separately?

Reply:
Done. Needed to be moved anyways since it seems deity needs the same treatment, and likely NPC as well.